src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -103,6 +104,11 @@ Copyright (c) .NET Foundation. All rights reserved.
     </RebuildDependsOn>
   </PropertyGroup>
 
+  <Target Name="_CheckForBuildWithNoBuild"
+          Condition="'$(NoBuild)' == 'true' and '$(PackAsTool)' != 'true'">
```

Comment:
What's the deal with PackAsTool here?

Reply:
That's a holdover from a different approach I was using that wasn't working for packing tools due to the dependency on the Publish target.  I will remove.

Will and I need to talk about removing pack tool's dependency on the Publish target (seems like it could now rely on the "copy local assets for build" behavior in the SDK).